roles/sushy_emulator/tasks/collect_details.yml

```diff
@@ -0,0 +1,86 @@
+---
+# Copyright Red Hat, Inc.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+- name: Get ingresses domain
+  environment:
+    KUBECONFIG: "{{ cifmw_openshift_kubeconfig }}"
+    PATH: "{{ cifmw_path }}"
+  ansible.builtin.command:
+    cmd: 'oc get ingresses.config/cluster -o jsonpath={.spec.domain}'
+  register: _ingress_domain
+  when: cifmw_sushy_emulator_install_type == 'ocp'
+
+- name: Base64 encode ssh private key
+  ansible.builtin.slurp:
+    src: "{{ cifmw_sushy_emulator_sshkey_path }}"
+  register: _cifmw_sushy_emulator_private_key_b64
+  no_log: true
+
+- name: Base64 encode ssh public key
+  ansible.builtin.slurp:
+    src: "{{ cifmw_sushy_emulator_sshkey_path }}.pub"
+  register: _cifmw_sushy_emulator_public_key_b64
+  no_log: true
+
+- name: Run ssh-keyscan
+  ansible.builtin.shell:
+    cmd: |
+      set -o pipefail
+      ssh-keyscan -H {{ hostvars[cifmw_baremetal_hypervisor_target].ansible_host }} 2>/dev/null | base64 -w 0
+  register: _cifmw_sushy_emulator_ssh_known_hosts_b64
+
+- name: Write known hosts for later use
+  ansible.builtin.copy:
+    content: "{{ _cifmw_sushy_emulator_ssh_known_hosts_b64.stdout | b64decode }}"
+    dest: "{{ cifmw_sushy_emulator_resource_directory }}/known_hosts"
+    mode: '0644'
+
+- name: Get details for Libvirt driver
+  when: cifmw_sushy_emulator_driver == 'libvirt'
+  block:
+    - name: Set vars
+      ansible.builtin.set_fact:
+        _libvirt_uri: "qemu+ssh://{{ cifmw_sushy_emulator_libvirt_user }}@{{ hostvars[cifmw_baremetal_hypervisor_target].ansible_host }}/system"
+
+    - name: Get Libvirt instance UUIDs
+      ansible.builtin.shell:
+        cmd: |
+          set -o pipefail
+          virsh --connect={{ _libvirt_uri }} list --all --uuid --name | grep {{ cifmw_sushy_emulator_instance_node_name_prefix }} | cut -d' ' -f1
```

Comment:
libvirt_manager is exposing this - we may extract the task into a dedicated file so that you can access it if needed? (as a follow-up)

Reply:
Yes I will add a todo for this. I would like to just slurp this info but it's not exposed in the deploy_layout task file yet. I will add this in a follow up.